model/registry/network.yaml

```diff
@@ -141,12 +141,12 @@ groups:
       - id: protocol.version
         stability: stable
         type: string
-        brief: Version of the protocol specified in `network.protocol.name`.
-        examples: '3.1.1'
+        brief: The actual version of the protocol used for network communication.
+        examples: ['1.1', '2']
         note: >
-          `network.protocol.version` refers to the version of the protocol used and might be
-          different from the protocol client's version. If the HTTP client has a version
-          of `0.27.2`, but sends HTTP version `1.1`, this attribute should be set to `1.1`.
+          If protocol version is subject to negotiation (for example using [ALPN](https://www.rfc-editor.org/rfc/rfc7301.html)),
+          this attribute SHOULD be set to the negotiated version. If the actual protocol version is not known,
+          this attribute SHOULD NOT be set.
```

Comment:
so there will be no fallback on the initial version from request? It's not clear from issue if there was a consensus about this part

Reply:
I don't know enough about HTTP 😅, is there "an initial version", or do clients send a list of supported versions to the server and server decides?

Reply:
in case of HTTPS, the version negotiation happens before HTTP request starts as a part of TLS client hello.
So by the time application protocol kicks in, the version is already negotiated and known. Request line will contain the actual negotiated version.

If there was no negotiation, then request line would still contain the actual version. 

In case no request message was issued (DNS failed, connection was not established, protocol negotiation failed) - this is when no actual version is available. 

I believe what we agreed upon in the issue comments is that HTTP client configuration (default protocol or supported protocol list) should not be used on this attribute.

Reply:
> I believe what we agreed upon in the issue comments is that HTTP client configuration (default protocol or supported protocol list) should not be used on this attribute.

this makes sense to me, otherwise it would be hard to know if a given `network.protocol.version` represents the initial or negotiated version

if we find that the initial version is helpful in cases where no (negotiated) `network.protocol.version` was captured, I think we could introduce a new attribute for it(?)